server/src/Ogmios/Parser.hs

```diff
@@ -0,0 +1,146 @@
+  key strKey (withString rationalParser)
+
+parseExecutionPrices :: Parse Text.Text (Maybe ExecutionUnitPrices)
+parseExecutionPrices =
+  perhaps $
+    ExecutionUnitPrices
+      <$> parseRational "steps"
+      <*> parseRational "memory"
+
+parseExecutionUnits :: Parse e (Maybe ExecutionUnits)
+parseExecutionUnits = fromAesonParser
+
+parseCostModels :: Parse e (Map.Map AnyPlutusScriptVersion CostModel)
+parseCostModels =
+  Map.singleton (AnyPlutusScriptVersion PlutusScriptV1)
+    <$> key "plutus:v1" fromAesonParser
+
+parseResult :: Parse Text.Text ProtocolParameters
+parseResult =
+  key "result" $
+    ProtocolParameters
+      <$> parseVersion
+      <*> parseRational "decentralizationParameter"
+      -- TODO : How to parse the value `neutral` from ogmios?
+      <*> pure Nothing -- key "extraEntropy" (perhaps (makePraosNonce . BLU.fromString <$>asString))
+      <*> parseNatural "maxBlockHeaderSize"
+      <*> parseNatural "maxBlockBodySize"
+      <*> parseNatural "maxTxSize"
+      <*> parseNatural "minFeeConstant" -- I think minFeeConstant and minFeeCoefficient are swapped here
+      -- but this is consistent with the current config file.
+      <*> parseNatural "minFeeCoefficient"
+      <*> pure Nothing
+      <*> parseLovelace "stakeKeyDeposit"
+      <*> parseLovelace "poolDeposit"
+      <*> parseLovelace "minPoolCost"
+      <*> key "poolRetirementEpochBound" fromAesonParser
+      <*> parseNatural "desiredNumberOfPools"
+      <*> parseRational "poolInfluence"
+      <*> parseRational "monetaryExpansion"
+      <*> parseRational "treasuryExpansion"
+      <*> perhaps (parseLovelace "coinsPerUtxoWord")
+      <*> (key "costModels" parseCostModels <|> pure Map.empty)
+      <*> key "prices" parseExecutionPrices
+      <*> key "maxExecutionUnitsPerTransaction" parseExecutionUnits
+      <*> key "maxExecutionUnitsPerBlock" parseExecutionUnits
+      <*> perhaps (parseNatural "maxValueSize")
+      <*> perhaps (parseNatural "collateralPercentage")
+      <*> perhaps (parseNatural "maxCollateralInputs")
+
+decodeProtocolParameters :: ByteString -> Either [Text.Text] ProtocolParameters
+decodeProtocolParameters response =
+  let value :: Maybe Aeson.Value
+      value = Aeson.decode response
+   in case parseValue parseResult <$> value of
+        Just (Right params) -> Right params
+        Just (Left e) -> Left $ displayError id e
+        _ -> Left ["Fail at converting ogmios response to cardano format"]
+
+--      wrapped :: Maybe ProtocolParametersWrapper
+--      wrapped =
+--        encoded
+--          >>= Aeson.decode @ProtocolParametersWrapper
+--   in unwrapParams <$> wrapped
+
+type LocalParser a = Parsec.Parsec String () a
+
+nonZeroDigit :: LocalParser Char
+nonZeroDigit = Parsec.Char.satisfy (\c -> (c /= '0') && isDigit c)
+
+nonZeroInteger :: LocalParser Integer
+nonZeroInteger = do
+  headDigit <- nonZeroDigit
+  remains <- Parsec.many Parsec.Char.digit
+  return $ read (headDigit : remains)
```

Comment:
We favor `pure` over `return` everywhere else. There are several instances of this in this module and others